dapp-heroku/deploy-to-heroku.md

```diff
@@ -0,0 +1,112 @@
+
+This tutorial uses the source code from the tutorial, [Truffle: Adding a frontend with react box](https://github.com/kauri-io/kauri-fullstack-dapp-tutorial-series/tree/master/truffle-react-box-frontend). You will have to add a Procfile to the project, but no other changes are required.
+
+## Heroku
+
+Heroku is a Platform-as-a-Service (PaaS) that enables developers to quickly build, deploy, and scale web applications. We'll use Heroku to quickly deploy our application and make it accessible to the public.
+
+## Prerequisites
+
+You'll need to configure the application, as defined in an earlier tutorial, [Truffle: Adding a frontend with react box](https://kauri.io/article/86903f66d39d4379a2e70bd583700ecf/v14/truffle:-adding-a-frontend-with-react-box).
+
+Additionally, you'll need to install the Heroku CLI.
+
+More information on the Heroku CLI can be found [here](https://devcenter.heroku.com/articles/heroku-cli).
+
+The Heroku CLI is available through Homebrew:
+```bash
+brew tap heroku/brew && brew install heroku
+```
+
+## Setup Heroku
+
+Navigate to the project directory in your terminal. Note, you are going to need to run some `git` commands from the top level of the working tree. For the purposes of this tutorial, let's assume that you have an independent copy of the [react project from the previous tutorial](https://github.com/kauri-io/kauri-fullstack-dapp-tutorial-series/tree/master/truffle-react-box-frontend).
+
+If you haven't already, initialize your project as a git repository. Git is used natively by Heroku to update your code.
+
+```bash
+git init
+```
+
+Here we create a new Heroku app and indicate we want to use the standard nodejs buildpack by Heroku.
+
+```bash
+heroku create --buildpack heroku/nodejs
+```
+
+Heroku will automatically asign your application a random name. Alternatively, you may give you application a memorable name that will be accessible at your-app-name.herokuapp.com.
+
+```bash
+heroku create mynewdapp --buildpack heroku/nodejs
+```
+
+If you initialized the Git repository prior to running `heroku create` you'll be able to see the heroku remote by running `git remote`.
+
+If you don't see a remote named 'remote', you can add it manually.
+```
+git remote add heroku [your heroku git remote url here]
+```
+
+## Create a Procfile
+
+The only change required to the codebase to deploy your app is a Procfile. A Procfile specifies commands to be executed by the app upon startup. In this case, we use the Procfile to mimic the local deployment process, but on the Heroku server.
+
+Add a file called `Procfile` (no file extension) to the /client directory. The Procfile is specific to the React application, not the rest of the repository. Add a single line to the Procfile.
+
+```
+web: npm start
```

Comment:
i think `npm start` will run react app in development mode (which is slow)

Reply:
That's a good point, thanks for the feedback! Since React box is based on create-react-app, we should be able to use the [create-react-app buildpack](https://github.com/mars/create-react-app-buildpack). Are there any downsides to that in this context that you know of?

Reply:
i use `create-react-app-buildpack` too, i guess it ok

Reply:
Cool, thanks!